### PR TITLE
Add zig fmt support

### PR DESCRIPTION
**Pull request by arashm**
Not sure if I have done everything correctly, but tried to follow the other PRs, and tests pass as well. I tested it locally and it seems formatting files just fine.
Let me know of the issues or you can also edit the changes directly.

### CODE REVIEW

File: doc/ale-zig.txt [hunk 1]
```diff
@@ -7,6 +7,14 @@ Integration Information
 
   Currently, the only supported linter for zig is zls.
```
Comment (hsanson) [inline doc/ale-zig.txt:9]: Add a section separator here so the help index you added to the doc/ale.txt can properly jump here. The ale-zig-zigfmt tag must match the one in the index

```
===============================================================================
zigfmt                                                              *ale-zig-zigfmt*
```

File: doc/ale-zig.txt [hunk 2]
```diff
@@ -7,6 +7,14 @@ Integration Information
...
 
+g:ale_zig_zig_executable                             *g:ale_zig_zig_executable*
```
Comment (hsanson) [inline doc/ale-zig.txt:10]: For consistency better call these ziffmt..  If zig supports other tools in the future we can differentiate them if we use different name than the executable:

```
g:ale_zig_zigfmt_executable                                                 *g:ale_zig_zigfmt_executable*
```

File: doc/ale-zig.txt [hunk 3]
```diff
@@ -7,6 +7,14 @@ Integration Information
...
+                                                     *b:ale_zig_zig_executable*
```
Comment (hsanson) [inline doc/ale-zig.txt:11]: Same as above, lets use  `b:ale_zig_zigfmt_executable` instead.

**Review by hsanson — CHANGES_REQUESTED** [review]

File: doc/ale-zig.txt (hunk 1 above)
Comment (arashm) [reply doc/ale-zig.txt:9]: Done

File: doc/ale-zig.txt (hunk 2 above)
Comment (arashm) [reply doc/ale-zig.txt]: Done

File: doc/ale-zig.txt (hunk 3 above)
Comment (arashm) [reply doc/ale-zig.txt]: Done

**Comment by hsanson** [comment]
There are some linter issues, apply these changes to get them fixed:

```diff
diff --git a/doc/ale-supported-languages-and-tools.txt b/doc/ale-supported-languages-and-tools.txt
index 890a9091..738f1434 100644
--- a/doc/ale-supported-languages-and-tools.txt
+++ b/doc/ale-supported-languages-and-tools.txt
@@ -661,8 +661,8 @@ Notes:
 * Zeek
   * `zeek`!!
 * Zig
+  * `zigfmt`
   * `zls`
-  * `zig fmt`
 
 ===============================================================================
   vim:tw=78:ts=2:sts=2:sw=2:ft=help:norl:
diff --git a/doc/ale-zig.txt b/doc/ale-zig.txt
index 3c675bdc..cafa12a1 100644
--- a/doc/ale-zig.txt
+++ b/doc/ale-zig.txt
@@ -9,10 +9,10 @@ Integration Information
 
 
 ===============================================================================
-zigfmt                                                        *ale-zig-zigfmt*
+zigfmt                                                         *ale-zig-zigfmt*
 
-g:ale_zig_zigfmt_executable                      *g:ale_zig_zigfmt_executable*
-                                                 *b:ale_zig_zigfmt_executable*
+g:ale_zig_zigfmt_executable                       *g:ale_zig_zigfmt_executable*
+                                                  *b:ale_zig_zigfmt_executable*
   Type: |String|
   Default: `'zig'`
 
diff --git a/doc/ale.txt b/doc/ale.txt
index 97a9c361..3b445265 100644
--- a/doc/ale.txt
+++ b/doc/ale.txt
@@ -3285,8 +3285,8 @@ documented in additional help files.
   zeek....................................|ale-zeek-options|
     zeek..................................|ale-zeek-zeek|
   zig.....................................|ale-zig-options|
-    zls...................................|ale-zig-zls|
     zigfmt................................|ale-zig-zigfmt|
+    zls...................................|ale-zig-zls|
 
 
 ===============================================================================
diff --git a/supported-tools.md b/supported-tools.md
index 67f608da..592faad7 100644
--- a/supported-tools.md
+++ b/supported-tools.md
@@ -670,5 +670,5 @@ formatting.
 * Zeek
   * [zeek](http://zeek.org) :floppy_disk:
 * Zig
+  * [zigfmt](https://github.com/ziglang/zig)
   * [zls](https://github.com/zigtools/zls)
-  * [zig fmt](https://github.com/ziglang/zig)
```

**Comment by arashm** [comment]
Applied the changes form the linter now.

**Review by hsanson — APPROVED** [review]
Thanks. Looks good.

**Pull request merged**